### ISSUE TITLE
Error MySqlConnector.MySqlException (0x80004005)

### ISSUE THREAD

**Issue by brabaris** [suriyun-production/mmorpg-kit-docs#1030]
Already asked a similar question, but the error is not resolved.
The error MySqlConnector.MySqlException (0x80004005) often appears, after which some users cannot save their data even when they re-enter the game (ONLY RESTART THE SERVER). I consider this error to be critical and ask you to pay attention to it. Game server: Windows Server 2012 R2, Database: Ubuntu 20.04, MySQL 8.0 MMORPG KIT version: 1.65
I tried the game server + database on the local machine - everything is the same!

[Log_Database_Central_Chat_MapSpawn.txt](https://github.com/suriyun-production/mmorpg-kit-docs/files/6663415/Log_Database_Central_Chat_MapSpawn.txt)

https://youtu.be/Fp4nPcLU5Ks

**Comment by insthync** [comment]
It runs game servers separately from MySQL server?

**Comment by brabaris** [comment]
Yes, MySQL server side by side (both servers in the Netherlands). I tried everything on one server.

**Comment by insthync** [comment]
It seems like it occurring when set socket's no-delay option, https://github.com/mysql-net/MySqlConnector/blob/master/src/MySqlConnector/Core/ServerSession.cs (line 1035), But don't know how it's occurring.

**Comment by brabaris** [comment]
Maybe. But the error appears randomly, maybe on the third connection, maybe on the tenth, or maybe in 10 minutes ...
Tried MariaDB - same result.

**Comment by insthync** [comment]
As the log messages told, It seems like the problem not occurring at the MySQL or MariaDB side, not have to test on it. The weird thing is the problem is occurring just because it set no-delay option.

**Comment by brabaris** [comment]
Updated to 1.66b. There were many mistakes ... I especially liked the quest with DefaultInterestManager.cs.meta

**Comment by brabaris** [comment]
After the update, nothing has changed ...

**Comment by insthync** [comment]
As I said (this is third time), The problem occurs when set no-delay option, the MySQL Connector uses TCPClient to connect to MySQL server, I will try to change it, You actually should try to run database server on other platform,s not Windows 2012, or wait for me to do something.

**Comment by insthync** [comment]
Did you build the server as an IL2CPP build?

**Comment by brabaris** [comment]
1. The Database Server is now installed on a server running Ubuntu 20.04.
2. The server is built as mono.

**Comment by insthync** [comment]
Database server, you mean the database server which build from Unity not MySQL, right?

**Comment by brabaris** [comment]
Game server on Windows Server. And MySQL is on another server, on Ubuntu 20.04.
![10](https://user-images.githubusercontent.com/16080577/122369668-11137100-cf67-11eb-9dbd-c36cf4e27522.jpg)
![11](https://user-images.githubusercontent.com/16080577/122369676-12449e00-cf67-11eb-8462-5efbe4cd2c77.jpg)

**Comment by brabaris** [comment]
Suriyun, have any ideas?

**Comment by insthync** [comment]
Try runs game-server on other Windows or just runs on Linux

**Comment by insthync** [comment]
Or modify MySQLConnector

**Comment by brabaris** [comment]
Tried Win 2016 Server, Win 2019 Server - nothing changed (After a while, the error MySqlConnector.MySqlException (0x80004005) appears). :(
I started a game server on Ubuntu 20.04 and MySQL on the second server (also on Ubuntu 20.04) and everything seems to be ok!
Only I didn’t find where the logs are being written?
![10](https://user-images.githubusercontent.com/16080577/122549975-c57ec700-d03b-11eb-8802-5683c22a9f6a.jpg)
